React/Modules/RCTAppState.m

```diff
@@ -79,9 +81,18 @@ - (void)dealloc
 
 #pragma mark - App Notification Methods
 
-- (void)handleAppStateDidChange
+- (void)handleAppStateDidChange:(NSNotification *)notification
 {
-  NSString *newState = RCTCurrentAppBackgroundState();
+  NSString *newState;
+  
+  if (notification.name == UIApplicationWillResignActiveNotification) {
```

Comment:
It's unsafe to use == for string comparisons in Objective-C. It probably happens to work in this case because they are constants, and so the pointer is the same, but it's best practice to write this as:

    if ([notification.name isEqualToString:UIApplicationWillResignActiveNotification]) {